src/Arguments/Postgres/BasePostgresArguments.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace AzureMcp.Arguments.Postgres;
+
+public class BasePostgresArguments : SubscriptionArguments
+{
+    public string? User { get; set; }
+    public string? Server { get; set; }
+    public string? Database { get; set; }
+
+    protected static void ValidateProperties((string? Property, string Name)[] properties)
+    {
+        foreach (var (property, name) in properties)
+        {
+            if (string.IsNullOrEmpty(property))
+            {
+                throw new ArgumentNullException(name, $"{name} cannot be null or empty.");
```

Comment:
We don't throw exceptions here, we let it run through BindArguments and then show the missing params in the JSON returned